In [1]:
import pandas as pd
import matplotlib as plt
import seaborn as sns

In [4]:
df = pd.read_csv('breastcancerdf.csv')
df

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0          17.99         10.38          122.80     1001.0          0.11840   
1          20.57         17.77          132.90     1326.0          0.08474   
2          19.69         21.25          130.00     1203.0          0.10960   
3          11.42         20.38           77.58      386.1          0.14250   
4          20.29         14.34          135.10     1297.0          0.10030   
..           ...           ...             ...        ...              ...   
564        21.56         22.39          142.00     1479.0          0.11100   
565        20.13         28.25          131.20     1261.0          0.09780   
566        16.60         28.08          108.30      858.1          0.08455   
567        20.60         29.33          140.10     1265.0          0.11780   
568         7.76         24.54           47.92      181.0          0.05263   

     mean compactness  mean concavity  mean concave points  mean symmetry  \
0             0.27760         0.30010              0.14710         0.2419   
1             0.07864         0.08690              0.07017         0.1812   
2             0.15990         0.19740              0.12790         0.2069   
3             0.28390         0.24140              0.10520         0.2597   
4             0.13280         0.19800              0.10430         0.1809   
..                ...             ...                  ...            ...   
564           0.11590         0.24390              0.13890         0.1726   
565           0.10340         0.14400              0.09791         0.1752   
566           0.10230         0.09251              0.05302         0.1590   
567           0.27700         0.35140              0.15200         0.2397   
568           0.04362         0.00000              0.00000         0.1587   

     mean fractal dimension  ...  worst texture  worst perimeter  worst area  \
0                   0.07871  ...          17.33           184.60      2019.0   
1                   0.05667  ...          23.41           158.80      1956.0   
2                   0.05999  ...          25.53           152.50      1709.0   
3                   0.09744  ...          26.50            98.87       567.7   
4                   0.05883  ...          16.67           152.20      1575.0   
..                      ...  ...            ...              ...         ...   
564                 0.05623  ...          26.40           166.10      2027.0   
565                 0.05533  ...          38.25           155.00      1731.0   
566                 0.05648  ...          34.12           126.70      1124.0   
567                 0.07016  ...          39.42           184.60      1821.0   
568                 0.05884  ...          30.37            59.16       268.6   

     worst smoothness  worst compactness  worst concavity  \
0             0.16220            0.66560           0.7119   
1             0.12380            0.18660           0.2416   
2             0.14440            0.42450           0.4504   
3             0.20980            0.86630           0.6869   
4             0.13740            0.20500           0.4000   
..                ...                ...              ...   
564           0.14100            0.21130           0.4107   
565           0.11660            0.19220           0.3215   
566           0.11390            0.30940           0.3403   
567           0.16500            0.86810           0.9387   
568           0.08996            0.06444           0.0000   

     worst concave points  worst symmetry  worst fractal dimension  Class  
0                  0.2654          0.4601                  0.11890      0  
1                  0.1860          0.2750                  0.08902      0  
2                  0.2430          0.3613                  0.08758      0  
3                  0.2575          0.6638                  0.17300      0  
4                  0.1625          0.2364                  0.07678   

In [6]:
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix, classification_report

In [10]:
x = df.iloc[:, :-2]
y = df['Class']

In [11]:
x

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0          17.99         10.38          122.80     1001.0          0.11840   
1          20.57         17.77          132.90     1326.0          0.08474   
2          19.69         21.25          130.00     1203.0          0.10960   
3          11.42         20.38           77.58      386.1          0.14250   
4          20.29         14.34          135.10     1297.0          0.10030   
..           ...           ...             ...        ...              ...   
564        21.56         22.39          142.00     1479.0          0.11100   
565        20.13         28.25          131.20     1261.0          0.09780   
566        16.60         28.08          108.30      858.1          0.08455   
567        20.60         29.33          140.10     1265.0          0.11780   
568         7.76         24.54           47.92      181.0          0.05263   

     mean compactness  mean concavity  mean concave points  mean symmetry  \
0             0.27760         0.30010              0.14710         0.2419   
1             0.07864         0.08690              0.07017         0.1812   
2             0.15990         0.19740              0.12790         0.2069   
3             0.28390         0.24140              0.10520         0.2597   
4             0.13280         0.19800              0.10430         0.1809   
..                ...             ...                  ...            ...   
564           0.11590         0.24390              0.13890         0.1726   
565           0.10340         0.14400              0.09791         0.1752   
566           0.10230         0.09251              0.05302         0.1590   
567           0.27700         0.35140              0.15200         0.2397   
568           0.04362         0.00000              0.00000         0.1587   

     mean fractal dimension  ...  fractal dimension error  worst radius  \
0                   0.07871  ...                 0.006193        25.380   
1                   0.05667  ...                 0.003532        24.990   
2                   0.05999  ...                 0.004571        23.570   
3                   0.09744  ...                 0.009208        14.910   
4                   0.05883  ...                 0.005115        22.540   
..                      ...  ...                      ...           ...   
564                 0.05623  ...                 0.004239        25.450   
565                 0.05533  ...                 0.002498        23.690   
566                 0.05648  ...                 0.003892        18.980   
567                 0.07016  ...                 0.006185        25.740   
568                 0.05884  ...                 0.002783         9.456   

     worst texture  worst perimeter  worst area  worst smoothness  \
0            17.33           184.60      2019.0           0.16220   
1            23.41           158.80      1956.0           0.12380   
2            25.53           152.50      1709.0           0.14440   
3            26.50            98.87       567.7           0.20980   
4            16.67           152.20      1575.0           0.13740   
..             ...              ...         ...               ...   
564          26.40           166.10      2027.0           0.14100   
565          38.25           155.00      1731.0           0.11660   
566          34.12           126.70      1124.0           0.11390   
567          39.42           184.60      1821.0           0.16500   
568          30.37            59.16       268.6           0.08996   

     worst compactness  worst concavity  worst concave points  worst symmetry  
0              0.66560           0.7119                0.2654          0.4601  
1              0.18660           0.2416                0.1860          0.2750  
2              0.42450           0.4504                0.2430          0.3613  
3              0.86630           0.6869                0.2575          0.6638  
4            

In [12]:
y

0      0
1      0
2      0
3      0
4      0
      ..
564    0
565    0
566    0
567    0
568    1
Name: Class, Length: 569, dtype: int64

In [15]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 42, stratify = y)

In [16]:
scaler = StandardScaler()

In [17]:
x_train = scaler.fit_transform(x_train)
x_test = scaler.transform(x_test)

In [18]:
x_train

array([[-1.07200079, -0.6584246 , -1.0880801 , ..., -1.07899452,
        -1.35052668, -0.35265805],
       [ 1.74874285,  0.06650173,  1.75115682, ...,  1.17859444,
         1.54991557,  0.19107787],
       [-0.97473376, -0.93112416, -0.99770871, ..., -1.28224057,
        -1.70744192, -0.307734  ],
       ...,
       [ 0.39844772,  1.06867262,  0.50751384, ...,  1.45815049,
         1.53492543,  0.16164487],
       [ 0.85331409, -0.0380331 ,  0.9054796 , ...,  1.04414791,
         2.10455077,  0.31035897],
       [-0.91179628, -0.82431683, -0.87666079, ...,  0.27780272,
        -0.51332734, -0.50756857]])

In [22]:
x_test

array([[ 1.56851278,  2.16401585,  1.74286587, ...,  0.68856004,
         1.20214432, -0.14043064],
       [-0.84027641, -0.5970672 , -0.8741735 , ..., -1.06990779,
        -1.10124064, -0.81429139],
       [-0.07072262,  1.19138742,  0.03202768, ...,  0.28522046,
         0.57255843,  1.14997397],
       ...,
       [ 0.3583966 , -0.44935493,  0.48264098, ...,  0.77108239,
         1.63386036,  0.54117564],
       [-0.41401799,  1.07321761, -0.42314565, ...,  0.53093308,
         0.06139464, -0.1280378 ],
       [-0.6829327 , -0.69932953, -0.66607058, ..., -0.5914636 ,
        -0.23615965, -0.16211811]])

In [24]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.metrics import classification_report, confusion_matrix

In [26]:
lr = LogisticRegression()
lr.fit(x_train, y_train)

LogisticRegression()

In [28]:
y_predict = lr.predict(x_test)
print(classification_report(y_test, y_predict))

              precision    recall  f1-score   support

           0       0.98      0.98      0.98        42
           1       0.99      0.99      0.99        72

    accuracy                           0.98       114
   macro avg       0.98      0.98      0.98       114
weighted avg       0.98      0.98      0.98       114



In [33]:
from sklearn.metrics import accuracy_score

print(f"Model Accuracy: {accuracy_score(y_test, y_predict) * 100:.3f} %")

Model Accuracy: 98.246 %


# ==========================================================

**testing model on unstandardized data, i.e: orignal data**

In [37]:
x1_train, x1_test, y1_train, y1_test = train_test_split(x, y, test_size = 0.2, random_state = 42, stratify = y)

In [44]:
lr.fit(x1_train, y1_train)

C:\Users\HP\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


LogisticRegression()

In [48]:
y1_predict = lr.predict(x1_test)

In [55]:
print(classification_report(y1_test, y1_predict))
print(confusion_matrix(y1_test, y1_predict))

              precision    recall  f1-score   support

           0       0.97      0.93      0.95        42
           1       0.96      0.99      0.97        72

    accuracy                           0.96       114
   macro avg       0.97      0.96      0.96       114
weighted avg       0.97      0.96      0.96       114

[[39  3]
 [ 1 71]]


In [50]:
print(f"Model Accuracy: {accuracy_score(y1_test, y1_predict) * 100:.3f} %")

Model Accuracy: 96.491 %


**so without balancing data through standardscaler, accuracy is 96.5% and with standardscalar accuracy is 98.246**

# ========================================================

In [53]:
print(classification_report(y_test, y_predict))
print(confusion_matrix(y_test, y_predict))

              precision    recall  f1-score   support

           0       0.98      0.98      0.98        42
           1       0.99      0.99      0.99        72

    accuracy                           0.98       114
   macro avg       0.98      0.98      0.98       114
weighted avg       0.98      0.98      0.98       114

[[41  1]
 [ 1 71]]


In [62]:
svm = SVC()
svm.fit(x_train, y_train)
y2_predict = svm.predict(x_test)

print(classification_report(y_test, y2_predict))
print(confusion_matrix(y_test, y2_predict))

              precision    recall  f1-score   support

           0       0.98      0.98      0.98        42
           1       0.99      0.99      0.99        72

    accuracy                           0.98       114
   macro avg       0.98      0.98      0.98       114
weighted avg       0.98      0.98      0.98       114

[[41  1]
 [ 1 71]]
